demo/main.c

```diff
@@ -5,20 +5,13 @@ int unit_tests(int argc, char **argv);
 
 void ndraw(mp_int *a, const char *name)
 {
-   char *buf;
-   int size;
-
-   mp_radix_size(a, 10, &size);
-   buf = (char *)malloc((size_t) size);
-   if (buf == NULL) {
-      fprintf(stderr, "\nndraw: malloc(%d) failed\n", size);
-      exit(EXIT_FAILURE);
-   }
+   char *buf = NULL;
+   size_t written;
 
    printf("%s: ", name);
-   mp_to_decimal(a, buf, (size_t) size);
+   mp_to_radix(a, &buf, &written, 10);
    printf("%s\n", buf);
-   mp_to_hex(a, buf, (size_t) size);
+   mp_to_radix(a, &buf, &written,  16);
```

Comment:
this needs a `free()` before `buf` can be passed again

Reply:
What?
Argh, great, forgot to clean up `demo/*c`.
Sorry!

Mmh, `valgrind` can't detect that? Ah, it isn't used in the `valgrind` runs, that's why it slipped through and made that gaping hole in my foot.
Ouch!

Will check the docs if I can use a big hammer there instead of the little note it has now.
Won't help those, who don't read the documentation&mdash;the majority, I'm afraid&mdash;but will COAs.